src/ddscxx/include/org/eclipse/cyclonedds/topic/TopicTraits.hpp

```diff
@@ -20,7 +20,8 @@
 
 #include <vector>
 
-#include <org/eclipse/cyclonedds/topic/DataRepresentation.hpp>
+#include "org/eclipse/cyclonedds/topic/DataRepresentation.hpp"
+#include "org/eclipse/cyclonedds/topic/datatopic.hpp"
```

Comment:
This will now get the cyclone DDS headers (from `ddsi`, `ddsrt`) which I guess will eventually lead to `pedantic` errors in the generated code because of definitions like `struct sockaddr_storage addrs[];` in `ddsrt/sockets.h`

However, the fix for the inclusion of `<cstdint>`, `<array>`, `<string>` for string const work. I am not sure what's the best way to include the headers for `ddscxx_sertype` without pulling the headers from cyclone DDS core!

Also the header `basic_cdr_ser.hpp` for and `write`/`read`.. is not included.

Reply:
`basic_cdr_ser.hpp` is included indirectly (via TopicTraits.hpp from the top of my mind), so we should be good on that front.

Reply:
The builds don't show any warnings and I've built everything with `-Wpedantic` and GCC 11.1 myself. Before I go of and check everything (which I'll happily do), is the fact that warnings will occur a hunch or can you reproduce?

Reply:
@k0ekk0ek It was my hunch, now I have tried compiling with this change and I don't see any pedantic errors (which I was expecting) so this should be good

Reply:
At least in my environment, this works as expected (maybe some includes are coming through other transient include files)